File: blog-sever/server-jpa/src/main/kotlin/com/site/blog/model/entity/AdminUser.kt
package com.site.blog.model.entity


import jakarta.persistence.*
import org.hibernate.Hibernate
import java.io.Serial
import java.io.Serializable

/**
 *
 *
 * 后台管理员信息表
 *
 *
 */
@Entity
data class AdminUser(
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "id", nullable = false)
    open var id: Long? = null,

    /**
     * 管理员登陆名称
     */
    var  username: String  = "",

    /**
     * 管理员登陆密码
     */
    var  password: String  = "",

    /**
     * 管理员显示昵称
     */
    var  nickname: String? = null,

    /**
     * 是否锁定 0未锁定 1已锁定无法登陆
     */
    var  locked: Boolean? = null,
    var  uuid: String? = null,

    /**
     * 0  普通用户,1管理员
     */
    var  role: Int? = null,
    var  avatar: String? = null,
    var  email: String? = null,


    ) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
        other as AdminUser

        return id != null && id == other.id
    }

    override fun hashCode(): Int = javaClass.hashCode()

    @Override
    override fun toString(): String {
        return this::class.simpleName + "(id = $id , username = $username , password = $password , nickname = $nickname , locked = $locked , uuid = $uuid , role = $role , avatar = $avatar , email = $email )"
    }

}

File: BlogApi/BlogApi/Models/AdminUser.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BlogApi.Models;

[Table("admin_user")]
public class AdminUser
{
    [Key]
    [Column("id")]
    public string Id { get; set; } = Guid.NewGuid().ToString("N");

    [Column("username")]
    public string Username { get; set; } = string.Empty;

    [Column("password")]
    public string Password { get; set; } = string.Empty;

    [Column("nickname")]
    public string? Nickname { get; set; }

    [Column("locked")]
    public bool Locked { get; set; }

    [Column("role")]
    public short Role { get; set; }

    [Column("avatar")]
    public string? Avatar { get; set; }

    [Column("email")]
    public string? Email { get; set; }

    [Column("uuid")]
    public string? Uuid { get; set; }

    [Column("create_time")]
    public DateTime? CreateTime { get; set; }

    [Column("update_time")]
    public DateTime? UpdateTime { get; set; }
}
